package WorldBattle;

message WorldBattleSoldier {
	required uint32 id = 1;
	required uint32 value = 2;
}

message WorldBattleUser
{
	required uint32 uid = 1;
	required uint32 ts = 2;
	repeated WorldBattleSoldier soldiers = 3;
	optional string name = 4;
	optional string fig = 5;
}

message WorldBattleCity
{
	required uint32 id = 1;
	repeated WorldBattleUser users = 2;
}

message WorldBattleRank
{
	required uint32 uid = 1;
	required uint32 value = 2;
	required uint32 cid = 3;
	required string name = 4;
	required string fig = 5;
}

message WorldBattleProtect
{
	required uint32 uid = 1;
	required uint32 ts = 2;
}

message WorldBattle
{
	repeated WorldBattleCity citys = 1;
	repeated WorldBattleRank ranks = 2;
	repeated WorldBattleProtect pro_city = 3;
}